Issue #51: said_uno

AurumTheEnd opened the pull request.

AurumTheEnd committed:
Fixed missing assertion in game_test

AurumTheEnd committed:
Changed Game::play_card's signature

AurumTheEnd committed:
Added Penalty & GainedCard WSMessages

AurumTheEnd committed:
Implemented the said_uno behavior

AurumTheEnd committed:
Added Game::message_all_but variant, excluding one player

AurumTheEnd committed:
Fixed DRAW message being sent to everyone - excluded the drawee

AurumTheEnd committed:
Implemented skip behavior

AurumTheEnd commented:
`fn test_active_cards()` extremely rarely randomly fails.

mazanec202 commented:
> `fn test_active_cards()` extremely rarely randomly fails, see if you can spot a mistake in it

Are you testing me, or genuinely don't know? :D

AurumTheEnd commented:
> Are you testing me, or genuinely don't know? :D
It was too late for me to grasp where the problem may lie, hoped you could genuinely clear it up :D

AurumTheEnd committed:
Introduced script for testing until failure

Unfortunate that such a feature is not part of `cargo test` itself, I'm used to it in Java.

AurumTheEnd committed:
Added Debug trait to Game, Deck, ActiveCards

AurumTheEnd committed:
Added counter to endless_test.sh

AurumTheEnd committed:
Attempted to close onto the rare test failure in test_active_cards

AurumTheEnd committed:
Fixed the rare test failure in test_active_cards

AurumTheEnd commented:
Fixed in bc0e904, took me an hour to find.

AurumTheEnd commented:
@mazanec202 reviewable, mergable

AurumTheEnd committed:
Executed rust fmt

AurumTheEnd committed:
Fixed cargo clippy errors in domain backend

holubond commented on endless_test.sh:
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+while :
+do
+	if ! cargo test ; then
+	  break
+	fi
+done
```
Toto je hodně cool :D

AurumTheEnd replied:
Is it? :D
